controllers/actions.github.com/autoscalingrunnerset_controller.go

```diff
@@ -305,6 +318,203 @@ func (r *AutoscalingRunnerSetReconciler) deleteEphemeralRunnerSets(ctx context.C
 	return nil
 }
 
+func (r *AutoscalingRunnerSetReconciler) removeFinalizersFromPermissionResources(ctx context.Context, autoscalingRunnerSet *v1alpha1.AutoscalingRunnerSet, logger logr.Logger) (requeue bool, err error) {
+	requeue, err = r.removeFinalizersContainerModeKubernetesPermissions(ctx, autoscalingRunnerSet, logger)
```

Comment:
it's a little weird that we abstract resources related to kube container mode into `removeFinalizersContainerModeKubernetesPermissions` and leave everything else as a flat list.

Is it possible to refactor out a common method that takes 3 parameters `remove FinalizerFromResource(resourceType sting,  resource object, annotationName string) (bool, err)`?
Then we can just use it for all resources we try to remove finalizers.